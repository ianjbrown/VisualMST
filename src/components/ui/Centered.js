import classes from "./Centered.module.css";
function Centered(props) {
  return (
    <div data-testid="centered-div" className={classes.centered}>
      {props.children}
    </div>
  );
}

export default Centered;
