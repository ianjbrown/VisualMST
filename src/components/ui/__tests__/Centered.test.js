import { render, screen, cleanup } from "@testing-library/react";
import Centered from "../Centered";
import "@testing-library/jest-dom/";
import renderer from "react-test-renderer";

afterEach(() => {
  cleanup();
});

test("Centered component matches snapshot", () => {
  const tree = renderer.create(<Centered>Hello World</Centered>).toJSON();
  expect(tree).toMatchSnapshot();
});

test("Centered component loads and has correct className", () => {
  render(<Centered>Hello World</Centered>);
  const centeredDiv = screen.getByTestId("centered-div");
  expect(centeredDiv).toBeInTheDocument();
  expect(centeredDiv).toHaveClass("centered");
});

test("Centered component is correctly passed children", () => {
  render(
    <Centered>
      <h1>Hello World</h1>
    </Centered>
  );
  const centeredDiv = screen.getByTestId("centered-div");
  expect(centeredDiv).toContainHTML("<h1>Hello World</h1>");
});
