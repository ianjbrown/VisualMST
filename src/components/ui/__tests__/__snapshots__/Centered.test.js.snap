// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Centered component matches snapshot 1`] = `
<div
  className="centered"
  data-testid="centered-div"
>
  Hello World
</div>
`;
